core/src/main/java/in/testpress/util/PermissionHandler.kt

```diff
@@ -0,0 +1,105 @@
+package `in`.testpress.util
+
+import android.Manifest
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build
+import android.provider.Settings
+import androidx.core.app.ActivityCompat
+import com.google.android.material.snackbar.Snackbar
+
+// Enum class representing permissions with their descriptions
+enum class Permission(val permission: String, val description: String) {
+    CAMERA(Manifest.permission.CAMERA, "Camera"),
+    MICROPHONE(Manifest.permission.RECORD_AUDIO, "Microphone"),
+    WRITE_STORAGE(Manifest.permission.WRITE_EXTERNAL_STORAGE, "Storage"),
+    READ_STORAGE(Manifest.permission.READ_EXTERNAL_STORAGE, "Storage");
+
+    companion object {
+        // Function to get all permissions based on the device's SDK version
+        fun getAllPermissions(): List<Permission> =
+            when {
+                Build.VERSION.SDK_INT >= Build.VERSION_CODES.R -> {
+                    listOf(CAMERA, MICROPHONE, READ_STORAGE)
+                }
+                else -> {
+                    values().asList()
+                }
+            }
+    }
+}
+
+// Utility class for handling permissions
+class PermissionHandler {
+
+    fun checkPermissionsGranted(
```

Comment:
This method name doesn't say what it does properly. `checkPermissionsGranted` - I would expect it to return true or false.